flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManagerDriver.java

```diff
@@ -689,4 +690,54 @@ public void onStopContainerError(ContainerId containerId, Throwable throwable) {
                     throwable);
         }
     }
+
+    public String getConatainerCompletedCause(ContainerStatus containerStatus) {
+        String completeContainerMessage = containerStatus.getDiagnostics();
+        switch (containerStatus.getExitStatus()) {
+            case ContainerExitStatus.SUCCESS:
+                log.debug(
+                        "Executor for container {} exited because of a YARN event (e.g., "
+                                + "preemption) and not because of an error in the running job. Diagnostics: {}",
+                        containerStatus.getContainerId().toString(),
+                        completeContainerMessage);
+                break;
+            case ContainerExitStatus.PREEMPTED:
+                completeContainerMessage =
+                        String.format(
+                                "Container %s was preempted by yarn. Diagnostics: %s",
+                                containerStatus.getContainerId().toString(),
+                                completeContainerMessage);
+                break;
+            case ContainerExitStatus.INVALID:
+                completeContainerMessage =
+                        String.format(
+                                "Container %s was invalid. Diagnostics: %s",
+                                containerStatus.getContainerId().toString(),
+                                completeContainerMessage);
+                break;
+            case ContainerExitStatus.ABORTED:
+                completeContainerMessage =
+                        String.format(
+                                "Container %s killed by YARN for being released by the application or being 'lost' due to node failures etc. Diagnostics: %s",
+                                containerStatus.getContainerId().toString(),
+                                completeContainerMessage);
+                break;
+            case ContainerExitStatus.DISKS_FAILED:
+                completeContainerMessage =
+                        String.format(
+                                "Container %s is failed for threshold number of the nodemanager-local-directories or"
+                                        + " threshold number of the nodemanager-log-directories become bad. Diagnostics: %s",
```

Comment:
```suggestion
                                "Container %s is failed because threshold number of the nodemanager-local-directories or"
                                        + " threshold number of the nodemanager-log-directories have become bad. Diagnostics: %s",
```